module/documents/combatant.mjs

```diff
@@ -10,14 +10,70 @@ export default class Combatant5e extends Combatant {
 
   /* -------------------------------------------- */
 
+  /**
+   * Reset combat-related uses.
+   * @param {Set<string>} periods  Which recovery periods should be considered.
+   */
+  async recoverCombatUses(periods) {
+    /**
+     * A hook event that fires before combat-related recovery changes.
+     * @function dnd5e.preCombatRecovery
+     * @memberof hookEvents
+     * @param {Combatant5e} combatant     Combatant that is being recovered.
+     * @param {object} data
+     * @param {Set<string>} data.periods  Periods to be recovered.
+     * @returns {boolean}                 Explicitly return `false` to prevent recovery from being performed.
+     */
+    if ( Hooks.call("dnd5e.preCombatRecovery", this, { periods }) === false ) return;
```

Comment:
Not sure if this is maybe a consistent hook style we're trying to align on, but if not then I'm not sure of the benefit of an extra layer of nesting here.

```suggestion
    if ( Hooks.call("dnd5e.preCombatRecovery", this, periods) === false ) return;
```

Reply:
If more data ends up being passed through in the future, having an object to add the data to would make it a nonbreaking change. (Probably not in this case since it can just be appended to the tail of the arguments list but still.)

Reply:
Following that logic ad absurdum, all hook and function signatures should just take a single object argument. The combatant in this hook is not part of the data object, presumably because it's considered required information. It seems like the periods information is similarly important, and not really optional, but maybe not.